Dockerfile.cross

```diff
@@ -0,0 +1,297 @@
+WORKDIR /usr/src
+RUN git clone https://github.com/docker-archive/windows-container-utility.git containerutility
+
+FROM build-base-${BASE_VARIANT} AS containerutility-base-alpine
+ARG TARGETPLATFORM
+RUN xx-apk add --no-cache musl-dev gcc g++
+
+FROM build-base-${BASE_VARIANT} AS containerutility-base-bullseye
+ARG TARGETPLATFORM
+RUN xx-apt-get install -y binutils gcc g++ dpkg-dev pkg-config
+
+FROM containerutility-base-${BASE_VARIANT} AS containerutility-build
+ARG CONTAINERUTILITY_VERSION
+RUN --mount=from=containerutility-src,src=/usr/src/containerutility,rw \
+    --mount=type=cache,target=/root/.cache \
+    git fetch origin \
+    && git checkout -q "$CONTAINERUTILITY_VERSION" \
+    && CC="$(xx-info)-gcc" CXX="$(xx-info)-g++" make \
+    && mkdir /out \
+    && mv containerutility.exe /out/
+
+FROM binary-dummy AS containerutility-darwin
+FROM binary-dummy AS containerutility-freebsd
+FROM binary-dummy AS containerutility-linux
+FROM containerutility-build AS containerutility-windows-amd64
+FROM binary-dummy AS containerutility-windows-arm64
+FROM containerutility-windows-${TARGETARCH} AS containerutility-windows
+FROM containerutility-${TARGETOS} AS containerutility
+
+# vpnkit
+# TODO: build from source instead
+FROM scratch AS vpnkit-windows
+FROM scratch AS vpnkit-linux-386
+FROM djs55/vpnkit:${VPNKIT_VERSION} AS vpnkit-linux-amd64
+FROM scratch AS vpnkit-linux-arm
+FROM djs55/vpnkit:${VPNKIT_VERSION} AS vpnkit-linux-arm64
+FROM scratch AS vpnkit-linux-ppc64le
+FROM scratch AS vpnkit-linux-riscv64
+FROM scratch AS vpnkit-linux-s390x
+FROM vpnkit-linux-${TARGETARCH} AS vpnkit-linux
+FROM vpnkit-${TARGETOS} AS vpnkit
+
+FROM build-base-alpine AS build-alpine
+RUN apk add --no-cache lld
+ARG TARGETPLATFORM
+RUN xx-apk add --no-cache musl-dev gcc g++ pkgconf \
+    btrfs-progs-dev \
+    lvm2-dev \
+    libseccomp-dev \
+    libsecret-dev \
+    eudev-dev
+
+FROM build-base-bullseye AS build-bullseye
+RUN apt-get update && apt-get install -y lld
+ARG TARGETPLATFORM
+RUN xx-apt-get install --no-install-recommends -y binutils gcc g++ pkg-config dpkg-dev \
+    libapparmor-dev \
+    libbtrfs-dev \
+    libdevmapper-dev \
+    libseccomp-dev \
+    libsecret-1-dev \
+    libsystemd-dev \
+    libudev-dev
+
+FROM build-${BASE_VARIANT} AS build
+COPY --from=gowinres /out/ /usr/local/bin
+COPY --from=tini /out/ /out
+COPY --from=runc /out/ /out
+COPY --from=containerd /out/ /out
+COPY --from=rootlesskit /out/ /out
+COPY --from=containerutility /out /out
+COPY --from=vpnkit / /out
+ARG CGO_ENABLED
+ARG GO_LINKMODE
+ARG GO_LDFLAGS
+ARG GO_BUILDMODE
+ARG GO_BUILDTAGS="apparmor seccomp"
+ARG VERSION
+ARG PLATFORM
+ARG PRODUCT
+ARG DEFAULT_PRODUCT_LICENSE
+ARG PACKAGER_NAME
+RUN --mount=type=bind,target=.,ro \
+    --mount=type=tmpfs,target=cli/winresources/dockerd \
+    --mount=type=tmpfs,target=cli/winresources/docker-proxy \
+    --mount=type=cache,target=/root/.cache \
+    xx-go --wrap && OUTPUT=/out ./hack/build/release
```

Comment:
`xx-verify`

Reply:
It's done in release script: https://github.com/moby/moby/pull/43529/files#diff-5a394e70534b6f2ab7d8356d2d88802cba6c2a298de8273444244c68723ac08bR12-R22

Wanted to add it there but there is a little too much logic to use here docs in the dockerfile I think.

Reply:
I think people want to keep `xx` out of the scripts to the script can be used on their own on a manually prepared system.

Reply:
> I think people want to keep `xx` out of the scripts

I agree with you, that's why I check if `xx-verify` is available in the script so non-sandboxed invocation is still possible. But yeah `xx` purpose is to work inside a container. Will see how I can improve that.

Reply:
Moved `xx-verify` in the build stage.